src/Microsoft.AspNetCore.Routing/DispatcherMiddleware.cs

```diff
@@ -15,14 +15,14 @@ internal sealed class DispatcherMiddleware
     {
         private readonly MatcherFactory _matcherFactory;
         private readonly ILogger _logger;
-        private readonly IOptions<DispatcherOptions> _options;
+        private readonly CompositeEndpointDataSource _endpointDataSource;
```

Comment:
Also we need to have an issue for making `CompositeEndpointDataSource` consider data sources added into DI too
https://github.com/aspnet/Routing/blob/dev/src/Microsoft.AspNetCore.Routing/DependencyInjection/DispatcherServiceCollectionExtensions.cs#L25

Reply:
I think that is already done here https://github.com/aspnet/Routing/blob/dev/src/Microsoft.AspNetCore.Routing/DependencyInjection/ConfigureDispatcherOptions.cs